### PR TITLE
Allow merge command in other comments

### PR DESCRIPTION
**Pull request by Gijsvs**
This pull request add some more flexibility to the merge commands. With this you will be able to give a merge command at the end of a normal comment and trigger the bot. Examples of messages now marked as merge command:

> @somebody Looks good to me! @bot merge

> @bot merge
> You did some fine work here

Something that does not work yet is having interpunction at the end of a command. An example of this is `@bot merge. looks good`.

### CODE REVIEW

File: src/Logic.hs [hunk 1]
```diff
@@ -251,16 +251,19 @@ handlePullRequestClosed pr state = return $ Pr.deletePullRequest pr state {
...
     prefixCaseFold = Text.toCaseFold $ Config.commentPrefix config
+    -- Recursive function that checks a list of words, if the first element is
+    -- the bot prefix and the second merge that we have a merge command. otherwise
+    -- we recurse to check the rest of the message.
+    isMergeCommand' :: Text -> [Text] -> Bool
+    isMergeCommand' prefix messageWords =
+      case messageWords of
+        (p:c:_) | p == prefix && c == "merge" -> True
+        (_:xs)  -> isMergeCommand' prefix xs
+        _       -> False
```
Comment (ruuda) [inline src/Logic.hs]: So apart from space differences, this is similar to

```haskell
(prefixCaseFold <> " merge") `Text.isInfixOf` messageCaseFold
```

If you don't care about space, then I think the `isInfixOf` is a bit cleaner (and probably faster too). If you do care about the space, then we could use `List.isInfixOf`:

```haskell
[prefixCaseFold, "merge"] `List.isInfixOf` messageWords
```

File: tests/Spec.hs [hunk 2]
```diff
@@ -215,6 +215,38 @@ main = hspec $ do
...
+          pr     = fromJust $ Project.lookupPullRequest (PullRequestId 1) state'
+      Project.approvedBy pr `shouldBe` Just "deckard"
+
+    it "does not accepts merge command with interleaved comments" $ do
+      let state  = singlePullRequestState (PullRequestId 1) (Branch "p") (Sha "6412ef5") "sacha"
+          -- Note: "deckard" is marked as reviewer in the test config.
+          event  = CommentAdded (PullRequestId 1) "deckard" "@bot foo merge"
+          state' = fst $ runAction $ handleEventTest event state
+          pr     = fromJust $ Project.lookupPullRequest (PullRequestId 1) state'
+      Project.approvedBy pr `shouldBe` Nothing
```
Comment (ruuda) [inline tests/Spec.hs]: :+1: Thanks for adding tests!

File: src/Logic.hs (hunk 1 above)
Comment (Gijsvs) [reply src/Logic.hs]: I will change this, your suggestion is much cleaner. I guess I should have [hoogle'd](https://hoogle.haskell.org/?hoogle=%5Ba%5D+-%3E+%5Ba%5D+-%3E+Bool&scope=set%3Astackage) before trying to roll my own solution

**Comment by ruuda** [comment]
@OpsBotPrime merge

**Comment by OpsBotPrime** [comment]
Pull request approved by @ruuda, rebasing now.

**Comment by OpsBotPrime** [comment]
Rebased as d0bb0309985712d3d48e0f09a77a85c65c9b8200, waiting for CI …

**Pull request merged**